spyder/plugins/editor/plugin.py

```diff
@@ -2633,46 +2642,92 @@ def go_to_last_edit_location(self):
                 if position < editor.document().characterCount():
                     editor.set_cursor_position(position)
 
+    def _pop_next_cursor_diff(self, history, current_filename, current_cursor):
+        """Get the next cursor from history that is different from current."""
+        while history:

```

Comment:
Instead of doing a linear search on the list, you could create a dictionary and then just index it

Reply:
I am not sure what you mean. If you use a dictionary you loose the ordering, which is critical for an undo stack. On top of that this loop should mostly terminate after one iteration. Could you expand on your comment?

Reply:
My bad, I didn't see that before. This looks great